src/runtime/internal/utils.ts

```diff
@@ -72,6 +72,15 @@ export function subscribe(store, ...callbacks) {
 	return unsub.unsubscribe ? () => unsub.unsubscribe() : unsub;
 }
 
+export function subscribe_dynamic_store(store, callback) {
+	if (store == null) {
+		callback(undefined);
+		return noop;
+	}
+	const unsub = store.subscribe(callback);
+	return unsub.unsubscribe ? () => unsub.unsubscribe() : unsub;
+}
```

Comment:
[ask]

This is a question but why did you create the new function?
Why adding `callbacks.forEach(callback => callback(undefined));` to `subscribe` is not good?

Reply:
sorry didn't understand the question.

You are saying using what statement instead of which function?

Reply:
Already we have [subscribe](https://github.com/sveltejs/svelte/blob/master/src/runtime/internal/utils.ts#L67-L73) function.
And the difference between `subscribe_dynamic_store` and `subscribe` is only line number 77. (`callback(undefined);`)

So I thought that we can update `subscribe` without creating the new function  `subscribe_dynamic_store`.
But I believe you have a reason why `subscribe_dynamic_store` is necessary. So I would like to know about it.

Reply:
Hmm.. I can't remember why I did that.
I know we need to call `callback(undefined)` so that when `store` changed to undefined, we should update `$store` to undefined too.

The `subscribe` was used when it assumes that the store variable is static, so does not need to call `callback` to set the `$store` to undefined.

But let me look at it again and see if I can reuse the code.